pkg/scheduler/framework/plugins/tainttoleration/taint_toleration.go

```diff
@@ -167,3 +170,21 @@ func (pl *TaintToleration) ScoreExtensions() framework.ScoreExtensions {
 func New(_ context.Context, _ runtime.Object, h framework.Handle) (framework.Plugin, error) {
 	return &TaintToleration{handle: h}, nil
 }
+
+// isSchedulableAfterPodChange is invoked whenever a pod changed. It checks whether
+// that change made a previously unschedulable pod schedulable.
+func (pl *TaintToleration) isSchedulableAfterPodChange(logger klog.Logger, pod *v1.Pod, oldObj, newObj interface{}) (framework.QueueingHint, error) {
+	originalPod, modifiedPod, err := util.As[*v1.Pod](oldObj, newObj)
+	if err != nil {
+		return framework.Queue, err
+	}
+	if originalPod.Namespace != pod.Namespace ||
+		originalPod.Name != pod.Name ||
+		len(originalPod.Spec.Tolerations) == len(modifiedPod.Spec.Tolerations) {
```

Comment:
Only check `len()` isn't accurate in some scenarios. e.g., adds one toleration and remove one toleration at the same time.

Reply:
Can we do that(remove one toleration) during a update?
https://github.com/kubernetes/kubernetes/blob/0c645922edcc06adff43c70c02fb56751364bbb5/pkg/apis/core/validation/validation.go#L3857
I think most fields of it is immutable.

Reply:
Oh, so we can add toleration, but cannot delete existing toleration? I didn't know that. Then, checking just `len` seems fine.